lightning/src/routing/scoring.rs

```diff
@@ -377,6 +391,15 @@ impl<T: Time> Score for ScorerUsingTime<T> {
 			.and_modify(|failure| failure.add_penalty(failure_penalty_msat, half_life))
 			.or_insert_with(|| ChannelFailure::new(failure_penalty_msat));
 	}
+
+	fn payment_path_successful(&mut self, path: &[&RouteHop]) {
+		let half_life = self.params.failure_penalty_half_life;
+		for hop in path.iter() {
+			self.channel_failures
+				.entry(hop.short_channel_id)
+				.and_modify(|failure| failure.reduce_penalty(half_life));
```

Comment:
Hmm, it seems like it'd be useful to eventually have an `or_insert` that gives some incentive for using the hop (otherwise, seems like the positive feedback for the hop goes unused, iiuc?)

Reply:
I suppose there could be different ways to go about this, but note with the current interface we can't have negative channel penalties. That said, we could have this offset some of the other penalties that are applied, not only any accumulated `failure_penalty_msat`s as is what is done here simply by advancing the decay. We may want such offsets to decay as well. Otherwise, we could keep preferring a channel more and more, and then any future failures may not be enough to prefer a different channel until enough failures accumulate.

My inclination is to keep this simple for now, but I'm open to any specific way of implementing an offset.